src/extension.ts

```diff
@@ -417,6 +417,14 @@ export function activate(context: vscode.ExtensionContext) {
 
             const term = selection.selectTerm(editor.document, pos);
             if (!term) {
+                if (!editor.selection.isEmpty) {
```

Comment:
I think this condition should be checked before the term is selected (before line 418). Then the behavior of `repl_send_goal` will be the same as the behavior of `repl_send_statement`: if there is a selected text, then send it to the REPL as a new goal, otherwise automatically select a term and send it as a new goal.

Reply:
Updated as you suggested.